include/aikido/trajectory/Trajectory.hpp

```diff
@@ -72,7 +72,7 @@ class Trajectory
       double _t, int _derivative, Eigen::VectorXd& _tangentVector) const = 0;
 
   /// Trajectory metadata
-  TrajectoryMetadata metadata;
+  mutable TrajectoryMetadata metadata;
```

Comment:
At a glance, I think this isn't necessary to be mutable, but let me learn more about `TrajectoryMetadata` not to make stupid suggestions.

I learned that `TrajectoryMetadata` was introduced by #529. What's the purpose of this class? For now, it only includes `executorValidated` and it's set by `TrajectoryExecutor::validate()` (probably this is not to re-validate once it's validated?). However, isn't this vulnerable to store the result in a trajectory to the case that we validate a trajectory using an executor when the trajectory is already validated by a different executor? We can workaround this by setting the flag to false before whenever we validate a trajectory, but it also has an issue that executors will revalidate the same trajectory every time.

Also, I thought the validation logic is different per executers, but it isn't as of now. I just assume some derived trajectory executor class would require different logic. If not, it would be better to have the validity check code in a single function.